src/files/files.service.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  DeleteObjectCommand,
+  GetObjectCommand,
+  PutObjectCommand,
+  S3Client,
+} from '@aws-sdk/client-s3';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { v4 as uuidv4 } from 'uuid';
+import {
+  DeleteFileResponseDto,
+  GetFileUrlResponseDto,
+  UploadFileResponseDto,
+} from './dto';
+import { UploadFileDto } from './dto/upload-file.dto';
+
+@Injectable()
+export class FilesService {
+  private client: S3Client;
+  private bucketName = this.configService.get('AWS_S3_BUCKET');
+
+  constructor(private configService: ConfigService) {
+    const s3_region = this.configService.get('AWS_BUCKET_REGION');
+
+    if (!s3_region) {
+      throw new Error('AWS_BUCKET_REGION not found in environment variables');
+    }
+
+    this.client = new S3Client({
+      region: s3_region,
+      credentials: {
+        accessKeyId: this.configService.get('AWS_ACCESS_KEY'),
+        secretAccessKey: this.configService.get('AWS_SECRET_ACCESS_KEY'),
+      },
+      forcePathStyle: true,
+    });
```

Comment:
you would get troubles with testing class with the dependency instantiated in the constructor. in NestJS you can create own providers which are instantiated by the framework and stored in IoC. So, in your module, you can add to the providers array a new object with some injection token and `useFactory` method which should instantiate the S3Client. then you can inject that provider in this class using `@Inject(<injection_token_of_s3_client_provider>)` decorator